sdk/azcore/CHANGELOG.md

```diff
@@ -1,9 +1,11 @@
 # Release History
 
-## 1.10.1 (Unreleased)
+## 1.11.0 (Unreleased)
 
 ### Features Added
 
+* Added field `AllowInsecureAuth` to `azcore.ClientOptions` and dependent authentication pipeline policies.
```

Comment:
I'm not super happy with the name `AllowInsecureAuth`. It's really more like "allow authorized requests over HTTP". I like having the word "Insecure" in the name as it's an indicator that enabling this has potentially dangerous repercussions. I can live with the name but open to suggestions.

Reply:
AllowInsecureRequests seems a bit more general and better since that's our overall intent.

I'm curious if we want to mimic the Go setting's name for skipping certs, which is a similar idea.

So maybe `InsecureAuth` or `InsecureRequests`, similar to here: [tls.InsecureSkipVerify](https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/crypto/tls/common.go;l=669)

Reply:
Perhaps even InsecureSkipHTTPSCheck

Reply:
I like the idea of having "Insecure" as the first word.

Reply:
We'll go with `InsecureAllowCredentialWithHTTP`